bus-mapping/src/mock.rs

```diff
@@ -44,7 +42,12 @@ impl BlockData {
         let mut code_db = CodeDB::new();
 
         sdb.set_account(&geth_data.eth_block.author, state_db::Account::zero());
-        sdb.set_account(&geth_data.eth_tx.from, state_db::Account::zero());
+        for tx in geth_data.eth_block.transactions.iter() {
+            sdb.set_account(&tx.from, state_db::Account::zero());
```

Comment:
I see.  Previously I added some comments in the `gen_associated_ops` for `BeginTx` and `EndTx` regarding returning errors when accounts don't exist.
From this I see that using the mocking framework, the empty accounts will be in the StateDB.
So one option is: for all "touched" accounts, set them in the StateDB even if they are zero.  The other option is to only set accounts that are not zero.  This will be important for the non-mock version, where we will be querying account data via Geth RPC.  We just need to be aware and follow the same strategy in both places.

Reply:
Yeah, in my mind the `state_db` is decoupled with the MTP proof. If we allow account creation while parsing the trace, we need to maintain a newly created account map, then query their absence proof after parsing. However, all accessed account should already be known in the first round iteration (along with their MTP proof), so I assume `state_db` should be a well prepared one.